### DISCUSSION TITLE
Nuxt3: How to pass config to the Vue app?

### DISCUSSION THREAD

**Discussion by mayakokits** [nuxt/nuxt#20047] [Questions]
When I'm injecting a ref in a component, I get the following warning from Vue:

__[Vue warn]:__ injected property "contentDir" is a ref and will be auto-unwrapped and no longer needs .value in the next minor release.
To opt-in to the new behavior now, set `app.config.unwrapInjectedRef = true` (this config is temporary and will not be needed in the future.)

As described in the [Vue docs](https://vuejs.org/api/application.html#app-config), properties to `app.config` must be set before the app is mounted. Does Nuxt allow to pass arbitrary keys other than `compilerOptions` to the Vue app config?

I've tried both
```
defineNuxtConfig({
  vue: {
    unwrapInjectedRef: true,
  },
})
```
and
```
defineNuxtConfig({
  vite: {
    vue: {
      unwrapInjectedRef: true,
    },
  },
})
```
but it doesn't show up in `nuxtApp.vueApp.config`. Have I missed something or does Nuxt not support this?

**Comment by TheAlexLichter** [comment]
It is `app.vue.compilerOptions`. See [source](https://github.com/nuxt/nuxt/blob/8e9d7157a8d4fae6d2d0ae4598dd7c2ae14e5571/packages/schema/src/config/webpack.ts#L162)

**Reply by mayakokits** [reply]
Thanks for pointing me to the source! In my case, I'm building with Vite, but it makes no difference.

Looks like it only passes properties of `vue.compilerOptions` through to Vite. Now the property I need to set on the Vue app config is a sibling of `compilerOptions`, not a child.

So there doesn't seem to be a way to set it via the nuxt config?

**Reply by TheAlexLichter** [reply]
Whoops, you are right.

Yes, I think the best would be raising an issue 👍 
Shouldn't be too tricky to add.

**Reply by cernymatej** [reply]
Hi, was this ever resolved? I wanted to configure the [warn handler](https://vuejs.org/api/application.html#app-config-warnhandler) and then stumbled upon this discussion.
